### PR TITLE
Prune what sre-admins can do in OSD clusters.

### PR DESCRIPTION
**Pull request by jewzaam**
The goal of this is to drive all changes to config management.  It is expected that some updates may be required over time.

### CODE REVIEW

**Comment by aweiteka** [comment]
This looks reasonable but it's difficult to know what capabilities we might need given our lack of experience supporting v4. There are so many objects. What are some things an SRE cannot do? I'm not looking for a comprehensive list but it would be good to think this through and discuss here.

**Comment by robotmaxtron** [comment]
> What are some things an SRE cannot do? I'm not looking for a comprehensive list but it would be good to think this through and discuss here.

- Create cronjobs (This seems strange to me)
- Create CRDs
- deletecollection nodes (Will this interfere with upgrades?)
- Create/Manage rolebindings & roles
- Create pods? Not sure if I'm reading that right.

@jewzaam Does this seem correct? Both in identifying some of the new restrictions and intended restrictions?

**Comment by jewzaam** [comment]
Right.  SRE will be able to do very little.  I verified these changes allow us to cordon a node and drain it.  Haven't gone so far as to kill a node off, per existing SOPs that is done by deleting the instance in AWS not the node in the cluster.  Are there things that we should be able to do that we can't given the rbac in this PR?

**Comment by jewzaam** [comment]
@robotmaxtron @aweiteka any chance of a lgtm or do we need some more changes?

File: deploy/sre-authorization/01-osd-sre-admin.ClusterRole.yaml [hunk 1]
```diff
@@ -3,12 +3,15 @@ kind: ClusterRole
...
   name: osd-sre-admin
 rules:
+# SRE can get, list, and watch all resources
 - apiGroups:
   - ""
   resources:
   - '*'
   verbs:
-  - '*'
+  - get
```
Comment (lisa) [inline deploy/sre-authorization/01-osd-sre-admin.ClusterRole.yaml]: This (and the following 2 lines) appears to duplicate lines 20-22?

Comment (jewzaam) [reply deploy/sre-authorization/01-osd-sre-admin.ClusterRole.yaml]: Value for `apiGroup` is "" on line #8 and "*" on line 16.  They are not duplicates, though they could be merged.

Comment (lisa) [reply deploy/sre-authorization/01-osd-sre-admin.ClusterRole.yaml]: What I mean is does `*` cover `""`?

Comment (jewzaam) [reply deploy/sre-authorization/01-osd-sre-admin.ClusterRole.yaml]: Turns out `"*"` does cover `""`.  Removed `""`.

**Comment by jeremyeder** [comment]
I think SRE might be missing the capability to run ```oc debug node```.

```
$ oc debug node/ip-10-0-134-242.ap-southeast-2.compute.internal
Error from server (Forbidden): pods is forbidden: User "jeder@redhat.com" cannot create resource "pods" in API group "" in the namespace "default"
```

https://jira.coreos.com/browse/SREP-1699

**Pull request merged**